pkg/network/status/status.go

```diff
@@ -236,7 +236,7 @@ func (m *Prober) IsReady(ctx context.Context, ing *v1alpha1.Ingress) (bool, erro
 
 		podCtx, cancel := context.WithCancel(ingCtx)
 		podState := &podState{
-			successCount: 0,
+			pendingCount: int32(len(ipWorkItems)),
```

Comment:
We actually end up with n-1 calls due to this callsite:
https://github.com/knative/serving/blob/cdcdfed3dad030d6e8c19c5ad9dcc06521a30e59/pkg/network/status/status.go#L256-L260

Reply:
sorry, n+1 (so pendingCount ends up at -1) 🤔

Reply:
Reaching -1 is not the problem.

In the happy path, the N probe requests complete, `updateStates` is called N times.
In the last call, `atomic.AddInt32(&podState.pendingCount, -1) == 0` is true, `podCtx` is cancelled, `m.updateStates` is called one last time but it is a no-op because the pending count is `-1`.

In the cancellation case, `M` (0 <= `M` < `N`) probing calls have succeeded so far (`atomic.AddInt32(&podState.pendingCount, -1) == 0` was never true at this point). Cancellation happens (Pod is killed or Ingress was updated). The problem is that `updateStates` called from line 259 will only decrement `pendingCount` by 1 and not execute the `atomic.AddInt32(&podState.pendingCount, -1) == 0` branch.
I'll send a fix.

Reply:
Yeah ack the real "bug".  I noticed it mostly because I have added some logging to some of these paths and included the counts, and saw `-1`s 😬